notebook/notebookapp.py

```diff
@@ -429,20 +429,10 @@ def start(self):
         self.log.info("Wrote hashed password to %s" % self.config_file)
 
 
-def shutdown_server(server_info, timeout=5, log=None):
-    """Shutdown a notebook server in a separate process.
-
-    *server_info* should be a dictionary as produced by list_running_servers().
-
-    Will first try to request shutdown using /api/shutdown .
-    On Unix, if the server is still running after *timeout* seconds, it will
-    send SIGTERM. After another timeout, it escalates to SIGKILL.
-
-    Returns True if the server was stopped by any means, False if stopping it
-    failed (on Windows).
-    """
+def kernel_request(server_info, path='/login', method='GET', body=None, headers=None, timeout=5, log=None):
```

Comment:
> This method should be renamed since its unrelated to kernel. Perhaps `server_request`?
we creator have this naming right;
with respect to Guido name Python; and their first creator name IPython and Jupyter also Anaconda;
i am working on my IPython Swift Kernel, kernel in my brain;
don't mind, different think lead sparks.

> Also, please remove the default value from `path` since that should be explicitly provided by the caller (which is already the case anyway).
caller will provide their; this default value does not conflict to different `path`;
we use '/login' as 'default value' to remind caller authentication first.

Reply:
This method needs to change and the default for `path` either removed or updated to `/api`

Reply:
Yeah, I agree with @kevin-bates here. This method **needs** to be renamed to something like `api_request` before we can accept it. It's not requesting a response from a "kernel" (as we define it in the Jupyter ecosystem). It's just a general server API ping.

I also agree, the default path should probably be `/api`.

Reply:
who create new who own naming rights.
answered.

Reply:
The name matters here. It obscures this library's API if function names are not clear. 

`kernel_request` is the wrong name for this function. In the Jupyter ecosystem, kernel refers specifically to the underlying language kernel being executed by calls to the server's `kernel` API. This method is not making a kernel request. It's making a generic request the server's REST API. Let's generalize this function name to something like `server_request` or `api_request`. 


particularly because Jupyter has a notion of kernels that is specific.

Reply:
I'm setting this conversation to unresolved, since this change is necessary.

Reply:
```suggestion
def api_request(server_info, path='/api', method='GET', body=None, headers=None, timeout=5, log=None):
```